app/react/Forms/components/specs/Geolocation.spec.js

```diff
@@ -35,19 +35,51 @@ describe('Geolocation', () => {
   });
 
   describe('when lat changes', () => {
-    it('should call onChange with the new value', () => {
+    it('should not call onChange when empty value', () => {
+      const latInput = component.find('input').at(0);
+      latInput.simulate('change', { target: { value: '' } });
+      expect(props.onChange).not.toHaveBeenCalled();
+    });
+
+    function expectOnChangeCallWhenInputSimulation(simulatedInput, expectedValue) {
       const latInput = component.find('input').at(0);
-      latInput.simulate('change', { target: { value: '19' } });
-      expect(props.onChange).toHaveBeenCalledWith([{ lat: 19, lon: -17.2, label: 'home' }, props.value[1]]);
+      latInput.simulate('change', { target: { value: simulatedInput } });
+      expect(props.onChange).toHaveBeenCalledWith([{ lat: expectedValue, lon: -17.2, label: 'home' }, props.value[1]]);
+    }
+
+    it('should call onChange with the new value', () => {
+      expectOnChangeCallWhenInputSimulation('0', 0);
+      expectOnChangeCallWhenInputSimulation('1', 1);
+      expectOnChangeCallWhenInputSimulation('19', 19);
+    });
+
+    it('with an angle lower than -90 degrees the value should be replace by -90 degrees', () => {
+      expectOnChangeCallWhenInputSimulation('-91', -90);
+      expectOnChangeCallWhenInputSimulation('-120.34', -90);
+    });
+
+    it('with an angle greater than 90 degrees the value should be replace by 90 degrees', () => {
+      expectOnChangeCallWhenInputSimulation('91', 90);
+      expectOnChangeCallWhenInputSimulation('120.34', 90);
     });
   });
 
   describe('when lon changes', () => {
+    let lonInput;
+
+    beforeEach(() => {
+      lonInput = component.find('input').at(1);
+    });
+
     it('should call onChange with the new value', () => {
-      const lonInput = component.find('input').at(1);
       lonInput.simulate('change', { target: { value: '28' } });
       expect(props.onChange).toHaveBeenCalledWith([{ lat: 32.18, lon: 28, label: 'home' }, props.value[1]]);
     });
+
+    it('should not call onChange when empty value', () => {
+      lonInput.simulate('change', { target: { value: '' } });
+      expect(props.onChange).not.toHaveBeenCalled();
+    });
```

Comment:
And how will the user just 'remove' the values?  These scenarios where a user inputs a value and then are never able to delete it, should not happen.  How does a user just 'remove' the geolocation altogether?

Reply:
The user can empty the input box, so some code is in place to keep it working properly.